### DISCUSSION TITLE
The future of translating CC: Tweaked

### DISCUSSION THREAD

**Discussion by SquidDev** [cc-tweaked/CC-Tweaked#1871] [Announcements]
For the last few years, CC: Tweaked's translations have been managed via a self-hosted [Weblate](https://weblate.org/en-gb/) instance. The aim of this was to make it easier for contributors to submit translation changes, without having to go through the whole GitHub flow (fork, branch, commit, PR).

Due to various issues with my VPS provider, I'm in the process of shutting down the server that hosts our Weblate instance. I want to get a new instance set up ASAP, but felt it was probably worth getting some feedback from the community first:

 - How have people found Weblate? Has it made translations easier to work on?
 - Have people experience/recommendations with other translation software (i.e. Crowdin)?
 - Are there any other pain points working with translations right now? Is there anything I can do to help there?

I don't actually our translation software at all ( :squid: ← English native speaker :disappointed:), so external feedback would be incredibly useful!

**Comment by tonitch** [comment]
Hello, I'm deep diving into the project, I really love(d) computercraft and your fork is awesome. I wanted to contribute and as a French speaker I took a look at https://github.com/cc-tweaked/CC-Tweaked/blob/mc-1.20.x/CONTRIBUTING.md#translations and followed a thread to here.

My point is, I don't really know Weblate. Most project I contribute to just let people do pull request. Did you have stats about contribution through Weblate? Was there many contribution? If not, then my personal option is that you shouldn't bother?

I know Arch Linux team uses Transifex for their translations, and most of the time when I try to contribute, it's not so great. I go to the "untranslated" category to see that most of them are translated, and I still find untranslated things in the "translated" category.
Anyway, my opinion is that it's not necessary (particularly if it's time-consuming to maintain and if not many people use it) if you care about translation, you can make a request for help in the README, so most people will see this first and will know to contribute this way.

**Reply by SquidDev** [reply]
> I wanted to contribute and as a French speaker I took a look at https://github.com/cc-tweaked/CC-Tweaked/blob/mc-1.20.x/CONTRIBUTING.md#translations and followed a thread to here. [...] Most project I contribute to just let people do pull request. 

Oh, I definitely do accept PRs, and that's the right way to go for now if you're looking to contribute :).

So my worry about PRs as the only workflow is that there is some overhead in making contributions. One has to fork the repo, make changes, and submit a PR. If you just want to change a single string, that's maybe not worth it — the hope is Weblate (or equivalent) lower the barrier of translations.

Whether it actually helped, that's a good question! Looking at git logs[^1], aside from a couple of commits (8564c1e54b7dc8a674b2758351863cb278c07590, 06f35e4997451ed0010235478dec4f3e1bb4b9cc), all translations over the last four years have been via Weblate (45 commits, 39 contributors). I know when I first set it up, there was an initial flurry of translation changes, but it's hard to know how much that was due to Weblate, and how much just me encouraging people to contribute translations!

[^1]: `git shortlog --since 2020-06-18 -- projects/common/src/main/resources/assets/computercraft/lang/ src/main/resources/assets/computercraft/lang/`

**Comment by zyxkad** [comment]
TBH the Weblate sucks, really sucks. When I translating I have to wait 5 seconds on switching translation items, which makes me really disappointing.

Crowdin is pretty good, it has machine & historic translate suggestions, quick switch (less than 1s), clean structure and a nice overview list. However I don't know if you must pay for Crowdin.

**Reply by SquidDev** [reply]
> However I don't know if you must pay for Crowdin.

It's free for open source projects. I'm not thrilled about commercial it is (it's quite upselly, and wow they try to shove AI in your face), but it does seem the best option for now.

I've set up a project at https://crowdin.com/project/cc-tweaked/, so we'll see how this goes.

**Reply by germe-deb** [reply]
Actually yeah, weblate isn't very good, it's a pain to work with.

**Reply by germe-deb** [reply]
I'm using that weblate to contribute right now. I'm adding translations for spanish (there was almost no translations there)

**Reply by germe-deb** [reply]
Oh my bad, tried to say I'm using that crowding to contribute
I'm reviewing some already approved sentences. I am writing correct sentences for them (they are not correct)

**Reply by SquidDev** [reply]
> Actually yeah, weblate isn't very good, it's a pain to work with.
>
> Oh my bad, tried to say I'm using that crowding to contribute

Sorry, did you mean that Crowdin isn't very good, or weblate? Definitely would love to know why!

**Reply by germe-deb** [reply]
Sorry I tried to say two things

weblate sucks

I'm right now translating cc to spanish through weblate

also, it would be cool if you or someone else that speaks spanish reviews my editions, because I made some changes to the terminology, like for example instead of [Ordenador] i used [Computadora], that is the same but it sounds way nicer and it's similar to Computer in english

**Reply by germe-deb** [reply]
The only good thing about weblate has than crowdin doesn't (or it's hidden) is that you can open the main weblate pago and if you don't have any projects, it will show you projects that are not Completed so you can just pick one and start contributing. Crowdin doesn't have that, or at least I didn't found it anywhere.

the bad thing about crowdin is that they are putting a lot of AI stuff on it (comically, it doesn't even work for me lol) but it isn't than invasive tbh

**Reply by germe-deb** [reply]
Oh shoot I just open the cc crowdin and saw all the approvals, thank you squiddev!! but please review my changes to the already approved changes you made, I changed some words like ordenador->computadora or "sin cables"->"inalámbrico"

**Comment by SirEndii** [comment]
We (Advanced Peripherals) used crowdin for a very long time before I bricked the configuration (Don't have the time or nerve to fix this currently)
Integration was a bit tricky since I needed to map the language codes with the ones from mc, [see here](https://github.com/IntelligenceModding/AdvancedPeripherals/blob/dev/1.20.1/crowdin.yml)

But I can recommend it, crowdin is very easy to work with, and they offer an open source license which basically gives you every feature without any rate or word limitations

**Reply by SquidDev** [reply]
> Integration was a bit tricky since I needed to map the language codes with the ones from mc, [see here](https://github.com/IntelligenceModding/AdvancedPeripherals/blob/dev/1.20.1/crowdin.yml)

Hah, I'd forgotten about this message entirely and hadn't realised you could do it via the config file! Spent ages setting up the mapping in the UI :D:.

**Comment by germe-deb** [comment]
I was totally unaware that a weblate existed for computercraft.

**Comment by Lupus590** [comment]
Maybe a message of the day for how to contribute translations would help spread awareness that translations are wanted and how to get started with contributing them.

**Reply by germe-deb** [reply]
That would work really well probably
